tests/test_commands.py

```diff
@@ -1472,3 +1638,152 @@ def verify_reporting_row(row, time_date):
+            assert f"Month now: {start_month}" in logs
+            assert f"Month {months_to_test} months ago: {end_month}" in logs
+            assert f"CSV file name: {csv_1_name}" in logs
+            assert f"CSV file name: {csv_2_name}" in logs
+            assert "Sending email with the CSV." in logs
+
+            # Verify that the csv files are attached - two files, one for each unit
+            assert len(outbox[-1].attachments) == 2
+            for attachment, file_name in zip(outbox[-1].attachments, [csv_1_name, csv_2_name]):
+                assert attachment.filename == file_name
+                assert attachment.content_type == "text/csv"
+
+            # Check csv content
+            # retrieve the files from the email
+            csv_1 = outbox[-1].attachments[0].data
+            csv_2 = outbox[-1].attachments[1].data
+
+            # check that the header and summatory at the end is correct
+            assert "Project ID,Project Title,Project Created,Time Collected,Byte Hours" in csv_1
+            assert "Project ID,Project Title,Project Created,Time Collected,Byte Hours" in csv_2
+            usage = 100.0 * months_to_test * 2  # 2 projects
+            assert f"--,--,--,--,{str(usage)}" in csv_1
+            usage = 100.0 * months_to_test
+            assert f"--,--,--,--,{str(usage)}" in csv_2
+
+            # check that the content is correct
+            import re
+
+            csv_1 = re.split(",|\n", csv_1)  # split by comma or newline
+            csv_2 = re.split(",|\n", csv_2)
+
+            # Projects and data is correct
+            assert csv_1.count("public_project_id") == months_to_test
+            assert csv_1.count("second_public_project_id") == months_to_test
+            assert csv_1.count("unit2testing") == 0  # this project is not in the unit
+            assert csv_1.count("100.0") == months_to_test * 2
+
+            assert csv_2.count("public_project_id") == 0  # this project is not in the unit
+            assert csv_2.count("second_public_project_id") == 0  # this project is not in the unit
+            assert csv_2.count("unit2testing") == months_to_test
+            assert csv_2.count("100.0") == months_to_test
+
+            # Check that the months included in the report are the correct ones
+            # move start time to the first day of the month
+            start_collected_time = start_time.replace(
+                day=1, hour=0, minute=0, second=0, microsecond=0
+            )
+            for i in range(months_to_test):
+                check_time_collected = start_collected_time - relativedelta(
+                    months=i
+                )  # every month is included
+                assert f"{check_time_collected}" in csv_1
+                assert f"{check_time_collected}" in csv_1
+
+    # Test that the command works for 4 months from Jan 2022
+    start_time = datetime(2022, 1, 15)  # Mid Jan 2022
+    run_command_and_check_output(months_to_test=4, start_time=start_time)
```

Comment:
We need to hard code somewhere that this in fact collects the rows for 
- 2021-10-01
- 2021-11-01
- 2021-12-01
- 2022-01-01

If the number of months is `4`. 

And same for the others below.

Reply:
The  lines               
_assert f"{check_time_collected}" in csv_1
assert f"{check_time_collected}" in csv_1_

Check for that. You mean harcode to do something like:

`_assert "2021-10-01" in csv_1`

?

Reply:
I added the new asserts. I didn't add for the dates after and before, because the function is already checking that the rows are repeated _months_to_test_ times. So it would be redundant

Reply:
Looks good